### PR TITLE
Feature/better project page

### PR DESCRIPTION
**Pull request by jimCresswell**
Redesign the project page. See individual commit messages.

Also note a bug in the update functionality, reported in #137 .

### CODE REVIEW

**Comment by jimCresswell** [comment]
This is separate from the file tree work.

**Comment by jimCresswell** [comment]
Found some issues, don't merge.

**Comment by jimCresswell** [comment]
Fixed...

**Comment by jimCresswell** [comment]
Closing to carry on with the file tree work in this branch.

**Comment by jimCresswell** [comment]
Okay, re-opening with basic file tree work done. Files are arranged by directory clicking the directory name will toggle those files.

**Comment by jimCresswell** [comment]
@sponte Ready for review, although it will probably need us both. We have the data to do whatever we want with the UI but I'd suggest this is good enough for now.

File: lib/specifications/repositoryTypes/git.js [hunk 1]
```diff
@@ -261,7 +262,7 @@ function getAllFiles(tree) {
   return _getAllFiles(tree)
     .then(function (files) {
       return files.filter(function (f) {
-        return /\.(feature|md)/.test(f);
+        return (appConfig.regex.fileOfInterest).test(f);
```
Comment (sponte) [inline lib/specifications/repositoryTypes/git.js]: Can we move file filtering into project.js  rather at low level inside please?

File: public/javascript/project.js [hunk 2]
```diff
@@ -1,3 +1,55 @@
...
+  selectEl.select2();
+
+  // On change, reload the page with a new query param dictating target branch.
+  selectEl.on('change', function() {
+    window.location.href = window.location.pathname + '?branch=' + this.value;
+  });
+});
+
+// Hide/collapse all button logic.
+$(function() {
```
Comment (sponte) [inline public/javascript/project.js]: This does not expand on first click

File: routes/feature.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ var router = express.Router();
 
 var markdown = require('markdown').markdown;
 var getProjectData = require('../lib/specifications/project').getData;
-var getFileContents = require('../lib/specifications/repositoryTypes/git').getFileContents;
+var getFileContents = require('../lib/specifications/project').getFileContents;
```
Comment (sponte) [inline routes/feature.js]: :+1:

**Pull request merged**